support/cas-server-support-cassandra-ticket-registry/src/main/java/org/apereo/cas/TicketSerializer.java

```diff
@@ -0,0 +1,23 @@
+package org.apereo.cas;
+
+import org.apereo.cas.ticket.ServiceTicket;
+import org.apereo.cas.ticket.Ticket;
+import org.apereo.cas.ticket.TicketGrantingTicket;
+
+/**
+ * Abstraction layer for ticket serialization methods
+ *
+ * Currently used by {@link org.apereo.cas.dao.CassandraDao}
+ * so {@param T} should be something that can be written
+ * to Cassandra like {@link String}
+ */
+public interface TicketSerializer<T> {
+
+    T serializeTGT(Ticket ticket);
+
+    T serializeST(Ticket ticket);
+
```

Comment:
You'll need to merge the two together. From the API perspective, you're just dealing with a generic Ticket. Inside the implementation of that API, you can detect the type as you see fit. Let me know if you need a sample to demonstrate that. Your API needs to remain ticket-agnostic as much as possible.